crates/client/src/storage/fs.rs

```diff
@@ -118,6 +121,16 @@ impl PackageStorage for FileSystemPackageStorage {
         store(&self.package_path(&info.name), info).await
     }
 
+    async fn load_checkpoint(&self) -> Result<Option<SerdeEnvelope<MapCheckpoint>>> {
+        let checkpoint = load(&self.package_path("checkpoint")).await?;
```

Comment:
I don't think this is the correct place to put the checkpoint. This is the path for a package named `checkpoint`. 

This should probably be a new file path with its own function & hard-coded name like `pending_publish_path`.